### PR TITLE
[CSV Sniffer] Early out if things go wrong in dialect detection

### PR DESCRIPTION
**Pull request by pdet**

### CODE REVIEW

File: test/sql/copy/csv/test_csv_wrong_file.test_slow [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# load the DB from disk
+load __TEST_DIR__/test_csv_tpch.db
+
+statement ok
+CALL dbgen(sf=1)
+
+restart
+
+statement error
+from read_csv('lineitem.db')
```
Comment (Mytherin) [inline test/sql/copy/csv/test_csv_wrong_file.test_slow:16]: This should probably be `__TEST_DIR__/test_csv_tpch.db`

**Review by Mytherin — COMMENTED** [review]
Thanks! LGTM (aside from the pending CI failure ;) )

File: test/sql/copy/csv/test_csv_wrong_file.test_slow (hunk 1 above)
Comment (pdet) [reply test/sql/copy/csv/test_csv_wrong_file.test_slow:16]: Absolutely.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**